### DISCUSSION TITLE
Add a disabled state to all the elements that can be disabled

### DISCUSSION THREAD

**Discussion by nestor-andreu** [tsujan/Kvantum#851] [Ideas]
The request is to add a disabled state to all the elements that can actually be disabled, such as buttons, toolbar buttons, line edits, checks and radios, arrows, scrollbars and handles, etc (in the same way that slidercursor or tab-close have a disabled state). Also allow for the text color present in such elements to be customized. If such state is not in the SVG (for example: button-disabled, button-toggled-disabled, lineedit-disabled, etc.), then display can be as it is now (usually with transparency), but if it is present, then show it. Some elements (such as those that admit a toggled state like buttons, checks and radios) should be able to hace two disabled states (as I showed with the button above).
The reason for that is that disabled elements would be able to have other traits than just being dimmed (for example I have some themes where disabled elements are reddish or brown, such as [Simple](https://www.pling.com/p/1745401), [CyberHack](https://www.pling.com/p/1620049), [DeeplySkeuo](https://www.pling.com/p/1929393) and others. This way one could create even richer Kvantum themes.
Also, KDE Plasma, qt5ct and qt6ct color schemes allow for such distinction for disabled elements, so it would be nice to be able to incorporate this richness fully into Kvantum.
Thanks for the great job with this project!

**Comment by tsujan** [comment]
Making disabled states as they are has been intentional, for various reasons: preventing a too heavy SVG file, keeping the code manageable, having a more natural look,... Kvantum is very flexible, but too much flexibility would have its cost.

> Also, KDE Plasma, qt5ct and qt6ct color schemes allow for such distinction for disabled elements

Kvantum covers much more parameters and cases, but there should be a limit to their number.

Also, Kvantum isn't intended to copy GTK themes, although it's possible to make Kvantum themes that match some GTK themes. Qt and GTK are totally different toolkits.
